File: 4.8-bc/ts/stage-2-feature-first/src/features/commerce/repository.ts
import type { DbPool } from "../../shared/db.js";
import type { Order } from "./model.js";

export class Repository {
  constructor(private readonly db: DbPool) {}

  async create(o: Order): Promise<void> {
    await this.db.query(
      `INSERT INTO commerce_orders (id, user_id, total_cents, status, created_at)
       VALUES ($1, $2, $3, $4, $5)`,
      [o.id, o.userId, o.totalCents, o.status, o.createdAt],
    );
  }
}

File: 4.8-bc/go/stage-2-feature-first/app/commerce/repository.go
package commerce

import (
	"context"

	"github.com/jackc/pgx/v5/pgxpool"
)

type Repository struct {
	db *pgxpool.Pool
}

func NewRepository(db *pgxpool.Pool) *Repository {
	return &Repository{db: db}
}

func (r *Repository) Create(ctx context.Context, o Order) error {
	_, err := r.db.Exec(ctx,
		`INSERT INTO commerce_orders (id, user_id, total_cents, status, created_at)
		 VALUES ($1, $2, $3, $4, $5)`,
		o.ID, o.UserID, o.TotalCents, o.Status, o.CreatedAt)
	return err
}
